org.eclipse.m2e.pde.connector/src/org/eclipse/m2e/pde/connector/PDEMavenBundlePluginConfigurator.java

```diff
@@ -0,0 +1,174 @@
+								+ (isFelix(execution.getPlugin()) ? FELIX_MANIFEST_GOAL
+										: Arrays.toString(BND_MANIFEST_GOALS))
+								+ ".",
+						IMarker.SEVERITY_WARNING, location);
+				problems.add(problem);
+				break;
+			}
+		}
+		if (problems.size() > 0) {
+			this.markerManager.addErrorMarkers(request.getPom(), IMavenConstants.MARKER_LIFECYCLEMAPPING_ID, problems);
+		}
+		IProject project = request.getProject();
+		IMavenProjectFacade facade = request.getMavenProjectFacade();
+
+		IPath metainfPath = getMetainfPath(facade, executions, monitor);
+
+		PDEProjectHelper.addPDENature(project, metainfPath, monitor);
+	}
+
+	private boolean isFelixBundleGoal(MojoExecution execution) {
+		return FELIX_MANIFEST_GOAL.equals(execution.getGoal());
+	}
+
+	private boolean isBNDBundleGoal(MojoExecution execution) {
+		String executionGoal = execution.getGoal();
+		for (String goal : BND_MANIFEST_GOALS) {
+			if (goal.equals(executionGoal)) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	@Override
+	public void configureClasspath(IMavenProjectFacade facade, IClasspathDescriptor classpath, IProgressMonitor monitor)
+			throws CoreException {
+	}
+
+	@Override
+	public void configureRawClasspath(ProjectConfigurationRequest request, IClasspathDescriptor classpath,
+			IProgressMonitor monitor) throws CoreException {
+	}
+
+	private IPath getMetainfPath(IMavenProjectFacade facade, List<MojoExecution> executions, IProgressMonitor monitor)
+			throws CoreException {
+		IMaven maven = MavenPlugin.getMaven();
+		for (MojoExecution execution : executions) {
+			Plugin plugin = execution.getPlugin();
+			MavenProject mavenProject = facade.getMavenProject(monitor);
+			File location = maven.getMojoParameterValue(mavenProject, execution,
+					isBND(plugin) ? BND_PARAM_MANIFESTLOCATION : FELIX_PARAM_MANIFESTLOCATION, File.class, monitor);
+			if (location != null) {
+				return facade.getProjectRelativePath(location.getAbsolutePath());
+			}
+		}
+		return null;
+	}
+
+	private boolean isBND(Plugin plugin) {
+		return plugin != null && "bnd-maven-plugin".equals(plugin.getArtifactId());
+	}
+
+	private boolean isFelix(Plugin plugin) {
+		return plugin != null && "org.apache.felix".equals(plugin.getGroupId())
+				&& "maven-bundle-plugin".equals(plugin.getArtifactId());
+	}
+
+	@Override
+	public boolean hasConfigurationChanged(IMavenProjectFacade newFacade,
+			ILifecycleMappingConfiguration oldProjectConfiguration, MojoExecutionKey key, IProgressMonitor monitor) {
+		return false;
+	}
+
+	@Override
+	public AbstractBuildParticipant getBuildParticipant(IMavenProjectFacade projectFacade, MojoExecution execution,
+			IPluginExecutionMetadata executionMetadata) {
+		if ((isFelix(execution.getPlugin()) && isFelixBundleGoal(execution))
+				|| (isBND(execution.getPlugin()) && isBNDBundleGoal(execution))) {
+			return new MojoExecutionBuildParticipant(execution, true, true);
+		}
```

Comment:
```suggestion
		if ((isFelix(execution.getPlugin()) && isFelixManifestGoal(execution))) {
			return new MojoExecutionBuildParticipant(execution, true, true) {
				@Override
				public Set<IProject> build(int kind, IProgressMonitor monitor) throws Exception {
					SubMonitor subMonitor = SubMonitor.convert(monitor, 10);
					if (kind == FULL_BUILD || kind == CLEAN_BUILD) {
						deleteFelixManifestIncrementalData(subMonitor.split(1));
					} else {
						subMonitor.setWorkRemaining(9);
					}
					return super.build(kind, subMonitor.split(8));
					// The Manifest Mojo refreshes the IFile corresponding to the generated
					// MANIFEST.MF by using the AbstractEclipseBuildContext to write it.
					// -> No need to refresh the MANIFEST.MF-IFile 'manually'
				}

				@Override
				public void clean(IProgressMonitor monitor) throws CoreException {
					deleteFelixManifestIncrementalData(monitor);
				}

				private void deleteFelixManifestIncrementalData(IProgressMonitor monitor) throws CoreException {
					SubMonitor subMonitor = SubMonitor.convert(monitor, "Delete Felix-Manifest mojo data", 3);
					IMavenProjectFacade projectFacade = getMavenProjectFacade();
					File buildDir = new File(projectFacade.getMavenProject().getBuild().getDirectory());
					IResource targetFolder = MavenProjectUtils.getProjectResource(projectFacade.getProject(), buildDir);

					IFolder felixDataFolder = ((IFolder) targetFolder).getFolder("maven-bundle-plugin");
					if (!felixDataFolder.exists()) {
						felixDataFolder.refreshLocal(IResource.DEPTH_ZERO, subMonitor.split(1));
					}
					felixDataFolder.delete(true, false, subMonitor.split(2));
				}
			};
		} else if (isBND(execution.getPlugin()) && isBNDBundleGoal(execution)) {
			return new MojoExecutionBuildParticipant(execution, true, true);
		}
```

Reply:
This uses the new method `MavenProjectUtils.getProjectResource()` introduce in PR #470.